admin/featured.php

```diff
@@ -6,138 +6,158 @@
-
-
-            } // nothing selected
-            if (isset($_GET['go_back']) && $_GET['go_back'] === 'ON') {
-                zen_redirect(zen_href_link(FILENAME_PRODUCTS_PRICE_MANAGER, 'products_filter=' . $products_id . '&current_category_id=' . $_GET['current_category_id']));
-            } else {
-                zen_redirect(zen_href_link(FILENAME_FEATURED, (isset($_GET['page']) && $_GET['page'] > 0 ? 'page=' . $_GET['page'] . '&' : '') . (isset($new_featured) ? 'fID=' . $new_featured->fields['featured_id'] : '') . (isset($_GET['search']) ? '&search=' . $_GET['search'] : '')));
-            }
-            break;
-        case 'update':
-            $featured_id = zen_db_prepare_input($_POST['featured_id']);
-
-            $featured_date_available = ((zen_db_prepare_input($_POST['start']) === '') ? '0001-01-01' : zen_date_raw($_POST['start']));
-            $expires_date = ((zen_db_prepare_input($_POST['end']) === '') ? '0001-01-01' : zen_date_raw($_POST['end']));
+  switch ($action) {
+    case 'setflag':
+      if (isset($_POST['flag']) && ($_POST['flag'] === '1' || $_POST['flag'] === '0')) {
+        zen_set_featured_status((int)$_POST['id'], (int)$_POST['flag']);
+        zen_redirect(zen_href_link(FILENAME_FEATURED, zen_get_all_get_params(['action', 'fID']) . 'fID=' . $_POST['id'], 'NONSSL'));
+      }
+      break;
+    case 'insert':
+      if (empty($_POST['products_id'])) {
+        $messageStack->add_session(ERROR_NOTHING_SELECTED, 'caution');
+      } else {
+        $products_id = (int)$_POST['products_id'];
+
+        $featured_date_available_raw = zen_db_prepare_input($_POST['featured_date_available']);
+        if (DATE_FORMAT_DATE_PICKER != 'yy-mm-dd' && !empty($featured_date_available_raw)) {
+          $local_fmt = zen_datepicker_format_fordate();
+          $dt = DateTime::createFromFormat($local_fmt, $featured_date_available_raw);
+          $featured_date_available_raw = 'null';
+          if (!empty($dt)) {
+            $featured_date_available_raw = $dt->format('Y-m-d');
+          }
+        }
+        $featured_date_available = (date('Y-m-d') < $featured_date_available_raw) ? $featured_date_available_raw : '0001-01-01';
+        $expires_date_raw = zen_db_prepare_input($_POST['expires_date']);
+        if (DATE_FORMAT_DATE_PICKER != 'yy-mm-dd' && !empty($expires_date_raw)) {
+          $local_fmt = zen_datepicker_format_fordate();
+          $dt = DateTime::createFromFormat($local_fmt, $expires_date_raw);
+          $expires_date_raw = 'null';
+          if (!empty($dt)) {
+            $expires_date_raw = $dt->format('Y-m-d');
+          }
+        }
+        $expires_date = (date('Y-m-d') < $expires_date_raw) ? $expires_date_raw : '0001-01-01';
+
+        $db->Execute("INSERT INTO " . TABLE_FEATURED . " (products_id, featured_date_added, expires_date, status, featured_date_available)
+                      VALUES (" . (int)$products_id . ", now(), 1, '" . zen_db_input($featured_date_available) . "')");
```

Comment:
Somehow you dropped the $expires_date from the provided fields, but it is still specified in the insert query.

Reply:
I added it when merging other updates.